assembly/std/storage.ts

```diff
@@ -188,49 +287,71 @@ export function appendOf<T>(address: Address, key: T, value: T): void {
+ *
+ * @throws
+ * - [compilation only] if the `key` type are neither `string`, `StaticArray<u8>`, or `Args`.
  */
 export function has<T>(key: T): bool {
   return env.has(toDatastoreFormat(key));
 }
 
 /**
- * Checks if the (key, value) exists in the datastore
- * of the given address.
+ * Checks if the key-value pair exists in the datastore of the target address.
+ *
+ * @remarks Can only be called at smart contract generation time by the parent smart contract to the child's address.
+ * 
+ * @typeParam T - Can be either `string`, `Args`, or `StaticArray<u8>`.
+ * 
+ * @param address - The child smart contract `address`.
+ * @param key - The key to check for existence in the datastore.
  *
- * @typeParam T - `string`, `Args` or `StaticArray<u8>`
- * @param address -
- * @param key -
+ * @returns A boolean value indicating whether the key-value pair exists in the datastore or not.
  *
+ * @throws
+ * - if the contract at the given `address` does not exist
+ * - if the caller is not authorized to modify the datastore.
+ * - [compilation only] if the `key` type are neither `string`, `StaticArray<u8>`, or `Args`.
  */
 export function hasOf<T>(address: Address, key: T): bool {
   return env.hasOf(address.toString(), toDatastoreFormat(key));
 }
 
 /**
- *  Sets the executable bytecode of the callee's address.
+ * Sets the executable bytecode of the current contract.
  *
- * @privateRemarks
- * TODO: explains failure consequences.
+ * @param bytecode - The bytecode to be set. It should be a `StaticArray<u8>`.
  *
- * @param bytecode -
+ * @throws
+ * - a runtime error if the callee's address does not correspond to a smart contract in the ledger.
+ * - a runtime error if the caller lacks write permissions on the callee's address.
+ * - a runtime error if the callee's address is the same as the creator's address.
+ * - a runtime error if an error occurs while updating the bytecode in the speculative ledger.
  */
 export function setBytecode(bytecode: StaticArray<u8>): void {
   env.setBytecode(bytecode);
 }
 
 /**
- *  Sets the executable bytecode of the given address.
+ * Sets the executable bytecode of the given `address`.
  *
- * @privateRemarks
- * TODO: explains security mechanisms.
+ * @remarks
+ * TODO: (add issue) explains security mechanisms.
+ *
+ * @param address - The target address whose bytecode will be set. It should be of type `Address`.
+ * @param bytecode - The bytecode to be set for the `address`. It should be a `StaticArray<u8>`.
  *
- * @param address - target address -
- * @param bytecode -
+ * @throws
+ * - a runtime error if the `address` does not correspond to a smart contract in the ledger.
+ * - a runtime error if the caller lacks write permissions on the `address`.
+ * - a runtime error if the `address` is the same as the creator's address.
+ * - a runtime error if an error occurs while updating the bytecode in the speculative ledger.
```

Comment:
@Ben-Rey I'm not sure about this part

Reply:
Same as https://github.com/massalabs/massa-as-sdk/pull/189#discussion_r1155646604